ArraysAndLists/Program.cs

```diff
@@ -18,46 +18,104 @@ public static class ArraysAndLists
         {
-            // Delete the below line when you are writing your code
-            return default;
+            if (names.Length == 0)
+            {
+                return null;
+            }
+            else if (names.Length == 1)
+            {
+                return names[0] + " liked your post.";
+            }
+            else if (names.Length == 2)
+            {
+                return names[0] + " and " + names[1] + " liked your post.";
+            }
+            else
+            {
+                return names[0] + ", " + names[1] + " and " + (names.Length - 2) + " others liked your post.";
+            }
         }
 
         // 2 - Write a method that accepts any number of integers (minimum of two integers). If an
         // integer is a duplicate, return "Error". If all integers are unique, then order them in
         // ascending order and return the sorted result as a string of integers seperated by spaces.
         public static string NumberSorter(params int[] numbers)
         {
-            // Delete the below line when you are writing your code
-            return default;
+            if (numbers.Length != numbers.Distinct().Count())
+            {
+                return "Error";
+            }
+            Array.Sort(numbers);
+
+            var result = string.Empty;
+
+            foreach (var number in numbers)
+            {
+                result += number + " ";
+            }
+            return result.Trim();
+
         }
 
         // 3 - Write a method that accepts a string for a name. Use an array to reverse the characters
         // then return the result as a string. "I like trains" should return "sniart ekil I". Return
         // "Error" if no text is inputted.
         public static string ReverseName(string name)
         {
-            // Delete the below line when you are writing your code
-            return default;
+            char[] backwardArray = name.ToCharArray();
+
+            if (backwardArray.Length < 1)
+            {
+                return "Error";
+            }
+            else
+            {
+                Array.Reverse(backwardArray);
+            }
+            return new string(backwardArray);
         }
 
         // 4 - Write a method that accepts any number of integers. If none or less than 5 numbers,
         // return "Invalid Array"; otherwise, return the 3 smallest numbers as a string seperated by
         // spaces, sorted in ascending order. Treat duplicate numbers normally.
         public static string SmallestNumbers(params int[] numbers)
         {
-            // Delete the below line when you are writing your code
-            return default;
+            if (numbers.Length < 5)
+            {
+                return "Invalid Array";
+            }
+            else
+            {
+                Array.Sort(numbers);
+                return numbers[0] + " " + numbers[1] + " " + numbers[2];
```

Comment:
Another way to do this is to use 

`numbers.Take(3)`

You can prob guess what that does.